storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java

```diff
@@ -98,13 +109,14 @@ public void storeGenesis(final UpdatableStore store) {
         // We need to store the genesis block in both hot and cold storage so that on restart
         // we're guaranteed to have at least one block / state to load into RecentChainData.
         // Save to hot storage
-        updater.addHotBlock(block);
+        hotUpdater.addHotBlock(block);
         // Save to cold storage
-        updater.addFinalizedBlock(block);
-        putFinalizedState(updater, root, state);
+        finalizedUpdater.addFinalizedBlock(block);
+        putFinalizedState(finalizedUpdater, root, state);
       }
 
-      updater.commit();
+      finalizedUpdater.commit();
```

Comment:
would commit failure here cause unhandled exceptions?

Reply:
It gets handled in `ChainStorage.onStorageUpdate` and even that isn't actually necessary as the event channel would have packaged it back up into the returned `SafeFuture` rather than leaving it unhandled.